File: client-api/src/main/java/org/ostelco/topup/api/health/HealthListener.java
package org.ostelco.topup.api.health;

/**
 * Interface for 'health' monitoring.
 */
@FunctionalInterface
public interface HealthListener {

    /**
     * Returns 'health state' of connection.
     * @return true if healthy
     */
    public boolean isHealthy();
}

File: client-api/src/main/kotlin/org/ostelco/prime/client/api/health/HealthListener.kt
package org.ostelco.prime.client.api.health

/**
 * Interface for 'health' monitoring.
 */
@FunctionalInterface
interface HealthListener {

    /**
     * Returns 'health state' of connection.
     * @return true if healthy
     */
    val isHealthy: Boolean
}
